pkg/query/storeset.go

```diff
@@ -89,13 +115,21 @@ func NewStoreSet(
 	if storeSpecs == nil {
 		storeSpecs = func() []StoreSpec { return nil }
 	}
-	return &StoreSet{
+
+	ss := &StoreSet{
 		logger:               logger,
 		storeSpecs:           storeSpecs,
 		dialOpts:             dialOpts,
 		storeNodeConnections: storeNodeConnections,
 		gRPCRetryTimeout:     3 * time.Second,
```

Comment:
can we init `externalLabelStores` just in case? If `Collect` run `externalLabelOccurrences` before the first `Update` we will get nil pointer exception.